File: Examen2Trimestre/Python/Rectangulo/Cuadrado.py
'''
Subclase de Rectangulo

@author: Cristina Prieto Jalao
'''
from Rectangulo import Rectangulo

class Cuadrado(Rectangulo):
      def __init__(self, alto):
          super().__init__(alto, alto);
          
      #Método para modificar el lado
      @lado.setter
      def lado(self, lado):
          self.alto=lado
          self.ancho=lado
      
      #Método para devolver el valor del lado del cuadrado
      @property
      def lado(self):
        return self.alto
      
      #Método para comparar los cuadrados
      def __gt__(self, cuadrado):
          return self.lado() > cuadrado.lado()

      def __lt__(self, cuadrado):
          return self.lado < cuadrado.lado()
  
      def __eq__(self, cuadrado):
          return self.lado == cuadrado.lado()
      
        
if __name__=="__main__": 
    #lado=int(input('Inserte el lado del cuadrado: '))
    cuadrado=Cuadrado(9)
    print(cuadrado)
    cuadrado2=Cuadrado(6)
    print(cuadrado2)

File: Examen2Trimestre/Java/Rectangulo/Cuadrado.java
package Rectangulo;

import java.util.Scanner;

/**
 * Subclase de Rectangulo
 * 
 * @author Cristina Prieto Jalao
 *
 */
public class Cuadrado extends Rectangulo implements Comparable<Cuadrado> {

  // Constructor
  public Cuadrado(int lado) {
    super(lado, lado);
  }

  public static void main (String [] args) {
    Scanner s=new Scanner (System.in);
    
    Cuadrado cuadrado = crearCuadrado(s);
    Cuadrado cuadrado2 = crearCuadrado(s);
    
    
    int resultado=cuadrado.compareTo(cuadrado2);
    if (resultado==0) {
      System.out.println("Los cuadrados son iguales");
    }
    else if(resultado>0) {
      System.out.println("El primer cuadrado es mayor que el segundo");
    }
    else 
      System.out.println("El primer cuadrado es menor que el segundo");
      
    
  }

  private static Cuadrado crearCuadrado(Scanner s) {
    Cuadrado cuadrado;
   
    do {
      try {
        System.out.println("Inserte el lado del cuadrado: ");
        cuadrado = new Cuadrado(s.nextInt());
        System.out.println(cuadrado);
        return cuadrado;
      } catch (Exception e) {
        System.out.println("El valor introducido es erróneo ");
      } 
    } while (true);
    
  }

  @Override
  public int compareTo(Cuadrado o) {
    if (alto == o.getAlto())
      return 0;

    if (alto > o.getAlto())
      return 1;
    
    return -1;
  
  }
}
